add Laravel ServiceProvider file and configuration

sandrokeil opened the pull request:
There are some problems with merging of configuration, because Laravel doesn't use `array_merge_recursive()` [here](https://github.com/laravel/framework/blob/5.2/src/Illuminate/Support/ServiceProvider.php#L67). For example the event_bus plugins are removed after merge.

A workaround is to have the full configuration in the user config files, but documentation says 

*You may also choose to merge your own package configuration file with the application's copy. This allows your users to include only the options they actually want to override in the published copy of the configuration*.

We provide a default package configuration and the user have to add some configuration to some sub keys. See [README.md](https://github.com/sandrokeil/prooph-package#example) for more details.

bweston92 commented:
@sandrokeil normally one would "publish" the config therefore all the defaults are there and they just update the defaults making a recursive merge not required.

sandrokeil commented:
Ok, that's a Laravel feature so I have to update the readme and it so it's the normal Laravel App behaviour. Thank's.

bweston92 commented:
Or you can use my PR to your repository? If not close it.

sandrokeil commented:
@bweston92 Is this [readme](https://github.com/sandrokeil/prooph-package/blob/master/README.md) understandable? Then we can merge it.

bweston92 commented:
@sandrokeil I just don't like how 2 service providers are required. I know it is a must to support container-interop, hmm. Other then that it is fine :+1:

sandrokeil commented:
@bweston92 Thanks for the response. The main reason is, that we can use the original factories with config validation. Maybe Laravel does support container-interop in the future. :-)

bweston92 commented:
@taylorotwell thoughts on Laravel with container-interop support without third-party package?

taylorotwell commented:
I don't have a big problem with it if someone wants to send over a PR.

bweston92 commented on src/ProophServiceProvider.php:
```diff
@@ -0,0 +1,106 @@
+            // prooph/event-store-bus-bridge set up
+            \Prooph\EventStoreBusBridge\TransactionManager::class,
+            \Prooph\EventStoreBusBridge\EventPublisher::class,
+            'Prooph\\EventStore\\Adapter\\Doctrine\\DoctrineEventStoreAdapter',
```
@sandrokeil what's wrong with `Prooph\EventStore\Adapter\Doctrine\DoctrineEventStoreAdapter::class` ?

bweston92 replied:
It won't attempt to load class mainly for performance reasons. It's just a constant: https://3v4l.org/qFEep

sandrokeil replied:
@bweston92 Good catch! I will fix it. I will fix it for dependencies too.

sandrokeil commented:
- Updated the code to use FCQN instead of strings for service names
- Moved `proophsoftware/prooph-cli` to suggest, it's a development tool
- Use only factories for services
- Supports `Zend\ServiceManager` v2 and v3

sandrokeil commented:
@bweston92 Are you working on a Laravel PR for container-interop?

bweston92 commented:
@sandrokeil not got time this week, busy week - sorry

codeliner commented on config/dependencies.php:
```diff
@@ -0,0 +1,31 @@
+ * @license   https://github.com/proophsoftware/prooph-package/blob/master/LICENSE.md New BSD License
+ */
+
+use Zend\ServiceManager\Factory\InvokableFactory;
```
Is this the only reason for the zend/service-manager dependency? If so we should provide our own InvokableFactory.

sandrokeil commented:
@codeliner I've updated the dependencies and added our own factory. This is a better solution. :+1:

codeliner commented:
thanks for the great work @sandrokeil and @bweston92 for review
